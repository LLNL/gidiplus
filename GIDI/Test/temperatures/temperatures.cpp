/*
# <<BEGIN-copyright>>
# Copyright 2019, Lawrence Livermore National Security, LLC.
# See the top-level COPYRIGHT file for details.
# 
# SPDX-License-Identifier: MIT
# <<END-copyright>>
*/

#include <stdlib.h>
#include <iostream>

#include "GIDI_testUtilities.hpp"

static char const *description = "This program prints.";

void main2( int argc, char **argv );
/*
=========================================================
*/
int main( int argc, char **argv ) {

    try {
        main2( argc, argv ); }
    catch (std::exception &exception) {
        std::cerr << exception.what( ) << std::endl;
        exit( EXIT_FAILURE ); }
    catch (char const *str) {
        std::cerr << str << std::endl;
        exit( EXIT_FAILURE ); }
    catch (std::string &str) {
        std::cerr << str << std::endl;
        exit( EXIT_FAILURE );
    }

    exit( EXIT_SUCCESS );
}
/*
=========================================================
*/
void main2( int argc, char **argv ) {

    argvOptions argv_options( "availableEnergy", description );
    ParseTestOptions parseTestOptions( argv_options, argc, argv );

    parseTestOptions.m_askGNDS_File = true;

    parseTestOptions.parse( );

    GIDI::Construction::PhotoMode photo_mode = parseTestOptions.photonMode( GIDI::Construction::PhotoMode::nuclearAndAtomic );
    GIDI::Construction::Settings construction( GIDI::Construction::ParseMode::all, photo_mode );
    PoPI::Database pops;
    GIDI::Protare *protare = parseTestOptions.protare( pops, "../pops.xml", "../all3T.map", construction, PoPI::IDs::neutron, "O16" );

    std::cout << stripDirectoryBase( protare->fileName( ), "/GIDI/Test/" ) << std::endl;

    std::vector<GIDI::Styles::TemperatureInfo> temperatures = protare->temperatures( );
    std::cout << "number of temperatures = " << temperatures.size( ) << std::endl;
    for( std::size_t i1 = 0; i1 < temperatures.size( ); ++i1 ) {
        GIDI::Styles::TemperatureInfo &temperatureInfo = temperatures[i1];
        GIDI::PhysicalQuantity temperature( temperatureInfo.temperature( ) );

        std::cout << "i1 = " << i1 << ":   temperature = " << temperature.value( ) << " " << temperature.unit( ) << std::endl;
        std::cout << "    heatedCrossSection = <" << temperatureInfo.heatedCrossSection( ) << ">" << std::endl;
        std::cout << "    griddedCrossSection = <" << temperatureInfo.griddedCrossSection( ) << ">" << std::endl;
        std::cout << "    heatedMultiGroup = <" << temperatureInfo.heatedMultiGroup( ) << ">" << std::endl;
        std::cout << "    SnElasticUpScatter = <" << temperatureInfo.SnElasticUpScatter( ) << ">" << std::endl;
    }

    delete protare;
}
