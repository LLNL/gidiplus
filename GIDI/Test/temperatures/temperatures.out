/.../GIDI/Test/Data/MG_MC3Ts/neutrons/n-008_O_016.xml
number of temperatures = 3
i1 = 0:   temperature = 2.586e-08 MeV/k
    heatedCrossSection = <heated_000>
    griddedCrossSection = <MonteCarlo_000>
    heatedMultiGroup = <MultiGroup_000>
    SnElasticUpScatter = <>
i1 = 1:   temperature = 1e-06 MeV/k
    heatedCrossSection = <heated_001>
    griddedCrossSection = <MonteCarlo_001>
    heatedMultiGroup = <MultiGroup_001>
    SnElasticUpScatter = <>
i1 = 2:   temperature = 0.0001 MeV/k
    heatedCrossSection = <heated_002>
    griddedCrossSection = <MonteCarlo_002>
    heatedMultiGroup = <MultiGroup_002>
    SnElasticUpScatter = <>
